python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -791,3 +794,114 @@ def airbyte_cloud_resource(context) -> AirbyteCloudResource:
 
     """
     return AirbyteCloudResource.from_resource_context(context)
+
+
+# ------------------
+# Reworked resources
+# ------------------
+
+
+@whitelist_for_serdes
+@record
+class AirbyteConnection:
+    """Represents an Airbyte connection, based on data as returned from the API."""
+
+    @classmethod
+    def from_connection_details(
+        cls,
+        connection_details: Mapping[str, Any],
+    ) -> "AirbyteConnection":
+        raise NotImplementedError()
+
+
+@whitelist_for_serdes
+@record
+class AirbyteDestination:
+    """Represents an Airbyte destination, based on data as returned from the API."""
+
+    @classmethod
+    def from_destination_details(
+        cls,
+        destination_details: Mapping[str, Any],
+    ) -> "AirbyteDestination":
+        raise NotImplementedError()
+
+
+@record
+class AirbyteWorkspaceData:
+    """A record representing all content in an Airbyte workspace.
+    Provided as context for the translator so that it can resolve dependencies between content.
+    """
+
+    connections_by_id: Mapping[str, AirbyteConnection]
+    destinations_by_id: Mapping[str, AirbyteDestination]
+
+
+@experimental
+class AirbyteClient:
```

Comment:
The workspace data is not specific to Airbyte Cloud - the REST API can be used [for OSS and Cloud instances](https://docs.airbyte.com/api-documentation) - which is why this object is named `AirbyteClient` .